examples/custom_distance/example_custom_distance.cpp

```diff
@@ -0,0 +1,142 @@
+/****************************************************************************
+ * Copyright (c) 2017-2024 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+#include <vector>
+
+constexpr int DIM = 2;
+using Coordinate = float;
+
+using Point = ArborX::ExperimentalHyperGeometry::Point<DIM, Coordinate>;
+using Box = ArborX::ExperimentalHyperGeometry::Box<DIM, Coordinate>;
+
+// Define custom type that will be used for distance calculations
+struct CustomPoint : public Point
+{};
+
+// Provide the required centroid function for the custom data type
+KOKKOS_FUNCTION auto returnCentroid(CustomPoint const &point)
+{
+  return Kokkos::bit_cast<Point>(point);
```

Comment:
Why did you chose to use `bit_cast` here, it is implicitly convertible?

Reply:
Done.